app.ts

```diff
@@ -0,0 +1,414 @@
+          firstId = finalId - 2;
+        } else if (window.innerHeight <= 170) {
+          finalId = 999999;
+          firstId = finalId - 1;
+        } else {
+          //pass
+        }
+      } else {
+        if (window.innerHeight > 500) {
+          amountRecord = 14;
+          finalId = firstId + amountRecord;
+        } else if (window.innerHeight <= 500 && window.innerHeight > 350) {
+          amountRecord = 10;
+          finalId = firstId + amountRecord;
+        } else if (window.innerHeight <= 350 && window.innerHeight > 170) {
+          amountRecord = 2;
+          finalId = firstId + amountRecord;
+        } else if (window.innerHeight <= 170) {
+          amountRecord = 1;
+          finalId = firstId + amountRecord;
+        } else {
+          //pass
+        }
+      }
+
+      if (firstId === 0) {
+        previousBtn.disabled = true;
+        nextBtn.disabled = false;
+      } else if (finalId === 999999) {
+        nextBtn.disabled = true;
+        previousBtn.disabled = false;
+      } else {
+        //pass
+      }
+
+      // To change the id so that the page does not cause an error
+      if (finalId > 999999 === true) {
+        let recordAmount = finalId - firstId;
+        nextBtn.disabled = true;
+        finalId = 999999;
+        firstId = finalId - recordAmount;
+
+        currentPage.innerHTML =
+          firstId +
+          " / " +
+          finalId +
+          " " +
+          (finalId - firstId + 1) +
+          " records only.";
+
+        console.log(typeof firstId, typeof finalId);
+
+        fetchDataScreenSize(firstId, finalId);
+      } else {
+        firstId = firstId;
+        finalId = finalId;
+
+        console.log(firstId, finalId);
+
+        currentPage.innerHTML =
+          firstId +
+          " / " +
+          finalId +
+          " " +
+          (finalId - firstId + 1) +
+          " records only.";
+
+        fetchDataScreenSize(firstId, finalId);
+      }
+    };
+
+    resizeScreen = debounce(resizeScreen, 700);
+    window.addEventListener("resize", resizeScreen);
+    resizeScreen();
+
+    firstPage.addEventListener("click", () => {
+      previousBtn.disabled = true;
+      nextBtn.disabled = false;
+      firstId = 0;
+
+      if (window.innerHeight > 500) {
+        finalId = firstId + 14;
+      } else if (window.innerHeight <= 500 && window.innerHeight > 350) {
+        finalId = firstId + 10;
+      } else if (window.innerHeight <= 350 && window.innerHeight > 170) {
+        finalId = firstId + 2;
+      } else if (window.innerHeight <= 170) {
+        finalId = firstId + 1;
+      } else {
+        //pass
+      }
+
+      resizeScreen = debounce(resizeScreen, 700);
+      window.addEventListener("resize", resizeScreen);
+      resizeScreen();
+    });
+
+    lastPage.addEventListener("click", () => {
+      previousBtn.disabled = false;
+      nextBtn.disabled = true;
+
+      finalId = 999999;
+
+      if (window.innerHeight > 500) {
```

Comment:
This checking for the screen size is being done for a second time (similar to earlier in the `nextPrevious()` function), so I think put this logic into its own method which you can call when you need to know whats the number of rows needed for the current screen size.